src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -529,28 +533,26 @@ bool VST3Wrapper::Initialize(EffectSettings& settings, Steinberg::Vst::SampleRat
       maxSamplesPerBlock,
       sampleRate
    };
-
-   //Set mActive before FetchSettings and IComponent::setActive
-   //to prevent redundant calls to AccessState::Flush.
-   mActive = true;
-   FetchSettings(settings);//Updates runtime settings...
-
+   
    if(!SetupProcessing(*mEffectComponent.get(), setup))
-   {
-      mActive = false;
       return false;
-   }
+
    mSetup = setup;
 
+   FetchSettings(settings);
+
    if(mEffectComponent->setActive(true) == kResultOk)
    {
-      //Omit the return value check - some plugins return unexpected values (see bug #3581)
-      mAudioProcessor->setProcessing(true);
-      //...make sure they are delivered to the processor
-      ConsumeChanges(settings);
-      return true;
+      if(mAudioProcessor->setProcessing(true) != kResultFalse)
```

Comment:
One suggestion, if setProcessing fails after setActive succeeds, should the failure path setActive back to false?